src/natcap/invest/validation.py

```diff
@@ -377,24 +337,19 @@ def check_vector(filepath, required_fields=None, projected=False,
     gdal.PopErrorHandler()
 
     if gdal_dataset is None:
-        return "File could not be opened as a GDAL vector"
+        return 'File could not be opened as a GDAL vector'
 
     layer = gdal_dataset.GetLayer()
     srs = layer.GetSpatialRef()
 
-    if required_fields:
-        fieldnames = set([defn.GetName().upper() for defn in layer.schema])
-        missing_fields = (
-            set(field.upper() for field in required_fields) - fieldnames)
-        if missing_fields:
-            return "Fields are missing from the first layer: %s" % sorted(
-                missing_fields)
+    if field_patterns:
+        fieldnames = [defn.GetName() for defn in layer.schema]
```

Comment:
I wonder who wrote this `layer.schema` fieldname lookup? I don't see a lot of documentation of using it directly but usually see `GetLayerDefn()` instead. Just a curiosity note.

Reply:
Huh yeah, not sure!